### ISSUE TITLE
run-duration assertion errors when running under FastAPI

### ISSUE THREAD

**Issue by rmasters** [timwie/aio-overpass#17]
Hey there. I'm running into some issues using aio-overpass in a FastAPI web application.

I'm using this sample, which works fine in a `asyncio.run(test_overpass())` script but runs into AssertionErrors with run_duration when run under FastAPI.

```python
async def test_overpass():
    query = Query('way["addr:housename"=Elbphilharmonie]; out geom;')

    client = Client()

    await client.run_query(query)

    return [elem.geojson for elem in collect_elements(query)]
```

Under FastAPI you get this:

```
  File ".../test.py", line 20, in main
    await client.run_query(query)
  File ".../.venv/lib/python3.12/site-packages/aio_overpass/client.py", line 243, in run_query
    await self._try_query_once(query)
  File ".../.venv/lib/python3.12/site-packages/aio_overpass/client.py", line 274, in _try_query_once
    req_timeout = _next_query_req_timeout(query)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../.venv/lib/python3.12/site-packages/aio_overpass/client.py", line 388, in _next_query_req_timeout
    assert run_duration
           ^^^^^^^^^^^^
AssertionError
```

<details>
<summary>Full FastAPI test case</summary>

```python
import logging

from fastapi import FastAPI

from aio_overpass import Client, Query
from aio_overpass.element import collect_elements

logging.basicConfig(level=logging.DEBUG)


app = FastAPI()


@app.get("/")
async def main():
    query = Query('way["addr:housename"=Elbphilharmonie]; out geom;')

    client = Client()

    await client.run_query(query)

    return [elem.geojson for elem in collect_elements(query)]
```

</details>

For info the state of the query looks like this:
```diff
- assert run-duration
+ assert run_duration, f"{run_duration=} {query.run_timeout_secs=} {query.run_timeout_elapsed=} {query.error=}"
```
```
AssertionError: run_duration=0.0 query.run_timeout_secs=None query.run_timeout_elapsed=False query.error=None
```

After an hour of poking around, unfortunately I wasn't able to track this down. To me it makes sense that the run-duration is 0.0 at that point as it appears to be called straight after `Query._begin_try`, so perhaps that assertion is incorrect? However it seems more likely there's an async-related problem as it works outside of FastAPI.

I'm using Python 3.12.3, `aio-overpass==0.14.0`, `aiohttp==3.10.2`, `fastapi==0.112.0`.

**Comment by timwie** [comment]
Yes, I think you're right, that assertion is wrong. It should be 

```python
assert run_duration is not None
```

instead of

```python
assert run_duration
```

That's unfortunate that no test ever ran into this. Seems like `asyncio.get_event_loop().time()` would need to return the same time twice for that to happen. 

Thank you for the report!

**Comment by timwie** [comment]
Should be fixed in 0.14.1 with db61438. Can you update and try again?

**Comment by timwie** [comment]
I'll close this as presumably fixed
